packages/arb-token-bridge-ui/src/tokenLists.tsx

```diff
@@ -157,34 +145,32 @@ export function fetchTokenLists(): Promise<void> {
           }
         })
 
-      storage = JSON.stringify(tokenListsWithBridgeTokenListId)
-      resolve()
+      resolve(tokenListsWithBridgeTokenListId as TokenListWithId[])
     })
   })
 }
 
 export function useTokenLists(forL2ChainId?: string): TokenListWithId[] {
-  const [tokenLists, setTokenLists] = useState<TokenListWithId[]>(() =>
-    getTokenLists(forL2ChainId)
+  const { data = [] } = useSWRImmutable(
+    ['useTokenLists', forL2ChainId],
+    async () => {
+      let newTokensList = []
+      const result = await fetchTokenLists()
+      if (typeof forL2ChainId === 'undefined') {
+        newTokensList = result
+      } else {
+        newTokensList = result.filter(
```

Comment:
Let's say we have
```
[
  { chainId: 2, ...},
  { chainId: 3, ...},
  { chainId: 4, ...},
]
```
and initially `forL2ChainId` set to `3`, if `chainId` changes, we would still fetch all tokensList again. Can we either:
- fetch all tokens list once, and filter (outside of the fetch function) based on `forL2ChainId`
- fetch only the list we need for a given `forL2ChainId` (reduced initial load)

Reply:
+1

Thinking we should also change 

```fetchTokenLists(forL2ChainId?: string)``` 

to 

```fetchTokenLists(forL2ChainId: number)```

That way we just fetch once we have the chain id. Inside the `useSWRImmutable` we can return `null` for the key while the chain id is `undefined`, which will prevent fetching and TypeScript will be happy.

Reply:
The only thing worth noting here is that we do need `fetchTokenLists` to function without a `chainId` as well (doesn't matter number or string, but should work in null case), the use case here is that during the start of the app, we fetch all the lists in `TokenListSyncer` through this function and then add it to the arbTokenBridge.

Reply:
I have still converted the handling of `chainId` from `string` to `number`.